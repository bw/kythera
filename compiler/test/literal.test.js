require("jest")
const t = require("./run")

describe("Literals", () => {
	t("int", "10")
	t("float", "3.14159")
	t("bool (true)", "true")
	t("bool (false)", "false")
	t("str", `"hello"`)
	t("null", "null")
	t("type", "int")
	t("type (type)", "type")
	t("fn", `
<int a> null {
	a
	return null
}`)
	t("obj", `
	{
		a = 10,
		b = "hello",
	}
	`)
})