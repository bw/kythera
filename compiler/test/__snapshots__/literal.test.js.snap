// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Literals bool (false) 1`] = `"new KYTHERA.value(false, KYTHERA.type.PRIMITIVES[\\"bool\\"]);"`;

exports[`Literals bool (true) 1`] = `"new KYTHERA.value(true, KYTHERA.type.PRIMITIVES[\\"bool\\"]);"`;

exports[`Literals float 1`] = `"new KYTHERA.value(3.14159, KYTHERA.type.PRIMITIVES[\\"float\\"]);"`;

exports[`Literals fn 1`] = `"new KYTHERA.value((a) => {a;return new KYTHERA.value(null, KYTHERA.type.PRIMITIVES[\\"null\\"]);}, new KYTHERA.type(\\"fn\\", { parameters: [KYTHERA.type.PRIMITIVES[\\"int\\"]], returns: KYTHERA.type.PRIMITIVES[\\"null\\"]}));"`;

exports[`Literals int 1`] = `"new KYTHERA.value(10, KYTHERA.type.PRIMITIVES[\\"int\\"]);"`;

exports[`Literals null 1`] = `"new KYTHERA.value(null, KYTHERA.type.PRIMITIVES[\\"null\\"]);"`;

exports[`Literals obj 1`] = `"new KYTHERA.value({\\"a\\": new KYTHERA.value(10, KYTHERA.type.PRIMITIVES[\\"int\\"]),\\"b\\": new KYTHERA.value(\\"hello\\", KYTHERA.type.PRIMITIVES[\\"str\\"]),}, new KYTHERA.type(\\"obj\\", {}));"`;

exports[`Literals str 1`] = `"new KYTHERA.value(\\"hello\\", KYTHERA.type.PRIMITIVES[\\"str\\"]);"`;

exports[`Literals type (type) 1`] = `"new KYTHERA.value(KYTHERA.type.PRIMITIVES[\\"type\\"], KYTHERA.type.PRIMITIVES[\\"type\\"]);"`;

exports[`Literals type 1`] = `"new KYTHERA.value(KYTHERA.type.PRIMITIVES[\\"int\\"], KYTHERA.type.PRIMITIVES[\\"type\\"]);"`;
